src/hdr_histogram.c

```diff
@@ -62,14 +67,20 @@ static void counts_inc_normalised(
     struct hdr_histogram* h, int32_t index, int64_t value)
 {
     int32_t normalised_index = normalize_index(h, index);
-    h->counts[normalised_index] += value;
-    h->total_count += value;
+    atomic_fetch_add_explicit(&h->counts[normalised_index], value, memory_order_relaxed);
+    atomic_fetch_add_explicit(&h->total_count, value, memory_order_relaxed);
 }
 
 static void update_min_max(struct hdr_histogram* h, int64_t value)
 {
-    h->min_value = (value < h->min_value && value != 0) ? value : h->min_value;
-    h->max_value = (value > h->max_value) ? value : h->max_value;
+    int64_t min = atomic_load_explicit(&h->min_value, memory_order_acquire);
+    if (value < min && value != 0) {
+        atomic_store_explicit(&h->min_value, value, memory_order_release);
```

Comment:
Note this isn't thread-safe - another thread might have already updated min_value between the load at line 76 and the store at line 78.

You should instead use a compare_exchange loop - see https://en.cppreference.com/w/c/atomic/atomic_compare_exchange

Reply:
Right miss understood the relationship between memory_order_aquire and memory_order_release will fix

Reply:
used a mutex as there's no atomic less than or greater than comparison and set